waku/v2/peermanager/peer_manager.go

```diff
@@ -73,40 +94,112 @@ func (pm *PeerManager) connectivityLoop(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-t.C:
-			pm.pruneInRelayConns()
+			pm.connectToRelayPeers()
 		}
 	}
 }
 
-func (pm *PeerManager) pruneInRelayConns() {
+// GroupPeersByDirection returns all the connected peers in peer store grouped by Inbound or outBound direction
+func (pm *PeerManager) GroupPeersByDirection() (inPeers peer.IDSlice, outPeers peer.IDSlice, err error) {
+
+	for _, p := range pm.host.Network().Peers() {
+		direction, err := pm.host.Peerstore().(wps.WakuPeerstore).Direction(p)
+		if err == nil {
+			if direction == network.DirInbound {
+				inPeers = append(inPeers, p)
+			} else if direction == network.DirOutbound {
+				outPeers = append(outPeers, p)
+			}
+		}
```

Comment:
Is it worth to add a `pm.logger.Error` line indicating the cause of error while attempting to retrieve the direction?